src/api/client.ts

```diff
@@ -81,6 +82,28 @@ export function getNostoClient(): PromiseLike<NostoClient> {
         }
     })
 }
+/**
+ * @group Nosto Client
+ * @category Core
+ */
+type LogLevel = 'error' | 'warn' | 'info' | 'debug'
+
+/**
+ * Logs an error to Nosto and the console
+ * @param msgOrError string message or error instance
+ * @param errorOrLevel error instance or log level string
+ * @param optLevel optional log level string
+ */
+export function log(msgOrError: string | unknown, errorOrLevel: LogLevel | unknown, optLevel?: LogLevel) {
+   const msg = typeof msgOrError === "string" ? msgOrError : undefined
+   const error = optLevel ? errorOrLevel : (!msg ? msgOrError : undefined)
+   const level = optLevel || (typeof errorOrLevel === "string" ? errorOrLevel : undefined) as LogLevel
+
+   getNostoClient().then(api => {
```

Comment:
only when error is defined

Reply:
the implementation below blurs the line between logging and error logging too much, since api.captureError expects an error, let's call it only when log is called with an error